static/js/TopicEditor.jsx

```diff
@@ -114,12 +114,52 @@ const TopicEditor = ({origData, onCreateSuccess, close, origWasCat}) => {
              .finally(() => setSavingStatus(false));
     }
 
+    const extractDisambiguationFromTitle = function(titleText){
+        let regex = /\((.+)\)$/;
+        let matches = titleText.match(regex);
+        let insideBrackets = matches ? matches[1] : null;
+        return insideBrackets
+    }
+    const removeDisambiguationFromTitle = function(titleText){
+        let regex = /\((.+)\)$/;
+        let newTitle = titleText.replace(regex, "");
+        return newTitle
+    }
+
+    const createPrimaryTitleObj = function(data, lang){
+        const fieldName = lang == 'en' ? 'enTitle' : 'heTitle';
+        let primaryTitleObj = {'text': removeDisambiguationFromTitle(data[fieldName]), "lang": lang, "primary": true};
+        let disambiguation = extractDisambiguationFromTitle(data[fieldName]);
+        if (disambiguation) {primaryTitleObj["disambiguation"]=disambiguation};
+        return primaryTitleObj;
+    };
+    const createNonPrimaryTitleObjArray = function(data, lang){
+        const fieldName = lang == 'en' ? 'enAltTitles' : 'heAltTitles';
+        const altTitles = data[fieldName];
+        const titleObjArray = []
+        if (Array.isArray(altTitles)) {
+            altTitles.forEach((title, index) => {
+                let titleObj = {'text': removeDisambiguationFromTitle(title['name']), "lang": lang};
+                let disambiguation = extractDisambiguationFromTitle(title['name']);
+                if (disambiguation) {titleObj["disambiguation"]=disambiguation}
+                titleObjArray.push(titleObj)
+            });
+        }
+        return titleObjArray
+    };
+
     const prepData = () => {
         // always add category, title, heTitle, altTitles
-        let postData = { category: data.catSlug, title: data.enTitle, heTitle: data.heTitle, altTitles: {}};
+        let postData = { category: data.catSlug, title: data.enTitle, heTitle: data.heTitle, altTitles: {}, titles: []};
         postData.altTitles.en = data.enAltTitles.map(x => x.name); // alt titles implemented using TitleVariants which contains list of objects with 'name' property.
```

Comment:
these lines can be removed and the data can be passed directly to createNonPrimaryTitleObjArray().